app/channel.go

```diff
@@ -1323,14 +1386,34 @@ func (a *App) RemoveUserFromChannel(userIdToRemove string, removerUserId string,
 		return err
 	}
 
+	var actorUser *model.User
 	if userIdToRemove == removerUserId {
 		a.postLeaveChannelMessage(user, channel)
 	} else {
+
+		if removerUserId != "" {
+			actorUser, err = a.GetUser(removerUserId)
+		}
+
+		if err != nil {
+			return err
+		}
+
 		a.Go(func() {
 			a.postRemoveFromChannelMessage(removerUserId, user, channel)
 		})
 	}
 
+	if a.PluginsReady() {
+		a.Go(func() {
+			pluginContext := &plugin.Context{}
+			a.Plugins.RunMultiPluginHook(func(hooks plugin.Hooks) bool {
+				hooks.UserHasLeftChannel(pluginContext, cm, actorUser)
+				return true
+			}, plugin.UserHasLeftChannelId)
+		})
+	}
```

Comment:
It looks like both `RemoveUserFromChannel` and `LeaveChannel` call `removeUserFromChannel`. Might be able to move this down a level into that function so it's only in one place

Reply:
Good call. I initially thought that this one doesnt pass the remover id, but that was apparently an oversight.